charts/metering-ansible-operator/templates/crds/meteringconfig.crd.yaml

```diff
@@ -159,6 +159,28 @@ spec:
                   type: string
                 secretName:
                   type: string
+              oneOf:
+              - required:
+                - enabled
+                allOf:
+                - not:
+                    required:
+                    - certificate
+                - not:
+                    required:
+                    - key
```

Comment:
I believe you can put this into the other `not` with `certificate`

Reply:
For some reason, that doesn't enforce the behavior we want, i.e.:
```
              oneOf:
              - required:
                - enabled
                allOf:
                - not:
                    required:
                    - certificate
                    - key
```
In the `MeteringConfig` CR:
```
  tls:
    enabled: false
    certificate: stub
    # key: stub
```
And applying that file:
```
[tflannag@localhost operator-metering]$ k create -f invalid_config.yaml 
meteringconfig.metering.openshift.io/operator-metering created
[tflannag@localhost operator-metering]$ 
```